cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/lineage/field/FieldLineageTable.java

```diff
@@ -481,4 +484,43 @@ private List<Field<?>> getOperationsKey(long checksum) {
     fields.add(Fields.longField(StoreDefinition.FieldLineageStore.CHECKSUM_FIELD, checksum));
     return fields;
   }
+
+  public List<EndPoint> getEndpoints(String namespaceId, ProgramRunId programRunId) throws IOException {
+    List<Range> multiRanges = new ArrayList<>();
+    multiRanges.add(getNamespaceIncomingRange(namespaceId));
+    multiRanges.add(getNamespaceOutgoingRange(namespaceId));
+    List<EndPoint> result = new ArrayList<>();
+    try (CloseableIterator<StructuredRow> iterator = getEndpointChecksumTable().multiScan(multiRanges,
+                                                                                          Integer.MAX_VALUE)) {
+      while (iterator.hasNext()) {
+        StructuredRow row = iterator.next();
+        ProgramRunId retrievedProgramRunId = GSON.fromJson(row.getString(
+          StoreDefinition.FieldLineageStore.PROGRAM_RUN_FIELD), ProgramRunId.class);
+        if (retrievedProgramRunId.equals(programRunId)) {
+          String namespace = row.getString(StoreDefinition.FieldLineageStore.ENDPOINT_NAMESPACE_FIELD);
+          String name = row.getString(StoreDefinition.FieldLineageStore.ENDPOINT_NAME_FIELD);
+          Map<String, String> properties = GSON.fromJson(
+            row.getString(StoreDefinition.FieldLineageStore.ENDPOINT_PROPERTIES_FIELD),
+            new com.google.gson.reflect.TypeToken<Map<String, String>>() { }.getType());
```

Comment:
this should be a private static final constant so that it's not created for every element in the iterator